### PR TITLE
feat(BRepHelper): add ray tracing based test for point in block.

### PR DESCRIPTION
**Pull request by francoisbonneau**

### CODE REVIEW

File: include/geode/model/helpers/ray_tracing.h [hunk 1]
```diff
@@ -47,4 +47,10 @@ namespace geode
             const BRep& brep,
             const Block3D& block );
 
+    bool opengeode_model_api test_point_inside_block(
```
Comment (BotellaA) [inline include/geode/model/helpers/ray_tracing.h:50]:
```suggestion
    bool opengeode_model_api is_point_inside_block(
```

File: include/geode/model/helpers/ray_tracing.h [hunk 2]
```diff
@@ -47,4 +47,10 @@ namespace geode
...
+        const BRep& brep, const Block3D& block, const Point3D& point );
+
+    absl::optional< uuid > opengeode_model_api block_containing_point(
```
Comment (BotellaA) [inline include/geode/model/helpers/ray_tracing.h:53]: What happens if the point is on a surface or on a line? How many uuid should we return?

File: src/geode/model/helpers/ray_tracing.cpp [hunk 3]
```diff
@@ -32,6 +32,55 @@
...
+        const geode::Block3D& block )
+    {
+        geode::index_t nb_intersections{ 0 };
+        const auto tracing =
+            find_intersections_with_boundaries( ray, brep, block );
+        for( const auto& intersections : tracing )
+        {
+            for( const auto& intersection : intersections.second )
+            {
+                if( intersection.position != geode::Position::inside )
```
Comment (BotellaA) [inline src/geode/model/helpers/ray_tracing.cpp:65]: If intersection point is on surface triangle edge or vertex, what is the expected behavior?

File: include/geode/model/helpers/ray_tracing.h (hunk 2 above)
Comment (francoisbonneau) [reply include/geode/model/helpers/ray_tracing.h:53]: if the point is on the boundary it is not in the block. 
Here the function return the block containing the point. 
Should we do a more generic function?

File: src/geode/model/helpers/ray_tracing.cpp (hunk 3 above)
Comment (francoisbonneau) [reply src/geode/model/helpers/ray_tracing.cpp:65]: Any advise? the problem is that if the intersection is on a vertex or an edge it is ambiguous in this context.

Comment (francoisbonneau) [reply src/geode/model/helpers/ray_tracing.cpp:65]: maybe a solution is to slightly change the direction of the ray untill we get a non ambigous response. the risk of infinit loop looks small don't you think?. 
we can either  chose the direction randomly or rotate the direction by a chosen increment.

**Review by BotellaA — APPROVED** [review]

**Pull request merged**